packages/compartment-mapper/src/node-modules.js

```diff
@@ -293,6 +342,13 @@ const graphPackage = async (
   } = {},
 ) => {
   if (graph[packageLocation] !== undefined) {
+    await updateShortestPaths(
+      graph,
+      packageLocation,
+      logicalPath,
+      preferredPackageLogicalPathMap,
+    );
```

Comment:
This probably obviates code farther down, run immediately before calling this function. https://github.com/endojs/endo/pull/2769/files#diff-047437f94608177a76b8636b72612ed583b0ae5a6d5cb619b851e7830c4425d8R575-R579

Reply:
@kriskowal I tried removing that bit, but it seems to be necessary.  I think because  `updateShortestPaths()` will only be called if we encounter the package two or more times.

Reply:
Can it trigger `updateShortestPaths` instead to reduce duplication?

Reply:
No, but I extracted what I could into another function to reduce duplication.